File: solutions/ruby/repeated_string.rb
def repeated_string s, n

	number_of_a = 0
	s_as_array = s.split('')
	number_of_repetitions = n % s_as_array.size === 0 ? (n / s_as_array.size) : ((n / s_as_array.size) + 1)
	last_for_n = n % s_as_array.size

	s_as_array.each do |letter|
		if letter === 'a'
			number_of_a = number_of_a + 1
		end	
	end

	occourences = number_of_a * number_of_repetitions

	unless last_for_n === 0
		occourences = occourences - number_of_a
		(0..(last_for_n - 1)).each do |idx|
			if s_as_array[idx] === 'a'
				occourences = occourences + 1
			end
		end
	end
	puts(occourences)
	return occourences
end

File: solutions/kotlin/repeatedString.kt
fun repeatedString(s: String, n: Long): Long {
    var numberOfAs = 0
    var occourences = 0L
    val numberOfRepetitions = if(n % s.length == 0L) (n / s.length) else ((n / s.length) + 1)
    var lastForN = n % s.length
    
    for (i in s.indices) {
        if(s[i] == 'a'){
            numberOfAs++
        }
    }
    
    occourences = numberOfAs * numberOfRepetitions
    
    if(lastForN != 0L){
        occourences = occourences - numberOfAs
        for(i in (0..(lastForN - 1))){
            if(s[i.toInt()] == 'a'){
                occourences++
            }
        }
    }
    
    return occourences
}